File: NotificationSystem/NotificationSystem.Web/ctrAgent.ascx.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Data;
using NotificationSystem.NotificationSystem.Data.NotificationSystemTableAdapters;
using static NotificationSystem.NotificationSystem.Data.NotificationSystem;
using NotificationSystem.NotificationSystem.Data.Classes;
using System.Diagnostics;
using System.Globalization;
using System.IO;
using System.Reflection;
using System.Runtime.CompilerServices;
using System.Security;
using System.Text;
using System.Threading.Tasks;
using Microsoft.VisualBasic;

namespace NotificationSystem.NotificationSystem.Web


{
    public partial class ctrAgent : System.Web.UI.UserControl
    {

        private int m_AgentID = 0;

        public int AgentID
        {
            get
            {
                return m_AgentID;
            }
            set
            {
                m_AgentID = value;
            }
        }
        protected void Page_Load(object sender, EventArgs e)
        {
            clsNotificationSystem theNotificationSystem = new clsNotificationSystem();
            AgentDataTable tblAgent = new AgentDataTable();
            int agentID = m_AgentID;

            try
            {
                if (Request.Form["ctl00$MainContent$ctrAgent_Search$btnSearch"] == "Search")
                    this.lblSearchResult.Text = Convert.ToString(m_AgentID);

                if ((Page.IsPostBack) & this.lblSearchResult.Text.Length > 0)
                    tblAgent = (AgentDataTable)theNotificationSystem.GetAgentByID(agentID);
                else
                    tblAgent = (AgentDataTable)theNotificationSystem.GetAgents();


                this.lblSearchResult.Text = tblAgent.Rows.Count + " Result(s)";
                this.grdAgents.DataSource = tblAgent.DefaultView;
                this.grdAgents.DataBind();
            }

            catch (Exception)
            {
                throw;
            }
        }


        private void grdAgents_PageIndexChanging(object sender, GridViewPageEventArgs e)
        {
            try
            {
                if (!(Information.IsNothing(ViewState["columnname"]) | Information.IsNothing(ViewState["direction"])))
                {
                    DataView m_DataView = (DataView)this.grdAgents.DataSource;

                    if (m_DataView == null)
                    {
                        m_DataView.Sort = ViewState["columnname"].ToString() + " " + ViewState["direction"].ToString();
                        this.grdAgents.DataSource = m_DataView;
                    }
                }

                this.grdAgents.PageIndex = e.NewPageIndex;
                this.grdAgents.DataBind();
            }
            catch (Exception)
            {
                throw;
            }
        }

        private void grdAgents_Sorting(object sender, GridViewSortEventArgs e)
        {
            try
            {
                DataView m_Dataview = (DataView)grdAgents.DataSource;

                if (m_Dataview == null)
                {
                    m_Dataview.Sort = e.SortExpression + " " + ConvertSortDirection(e);
                    this.grdAgents.DataSource = m_Dataview;
                    this.grdAgents.DataBind();
                }
            }
            catch (Exception)
            {
                throw;
            }
        }

        private string ConvertSortDirection(System.Web.UI.WebControls.GridViewSortEventArgs e)
        {
            ViewState.Add("columnname", e.SortExpression);

            if ((ViewState["direction"] == null))
                ViewState.Add("direction", "asc");
            else
                ViewState["direction"] = Interaction.IIf(ViewState["direction"].ToString().ToLower() == "desc", "asc", "desc");

            return ViewState["direction"].ToString();
        }
    }
}

File: NotificationSystem/NotificationSystem.Web/ctrAgent.ascx.vb
Imports System
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Data
Imports Microsoft.VisualBasic

Namespace NotificationSystem.NotificationSystem.Web
    Public Partial Class ctrAgent
        Inherits UserControl


        Private m_AgentID As Integer = 0

        Public Property AgentID As Integer
            Get
                Return m_AgentID
            End Get
            Set(value As Integer)
                m_AgentID = value
            End Set

        End Property
        Protected Sub Page_Load(sender As Object, e As EventArgs)
            Dim theNotificationSystem As NotificationSystem.NotificationSystem.Data.Classes.clsNotificationSystem = New NotificationSystem.NotificationSystem.Data.Classes.clsNotificationSystem()
            Dim tblGetAgent As NotificationSystem.NotificationSystem.Data.NotificationSystem.AgentDataTable = New NotificationSystem.NotificationSystem.Data.NotificationSystem.AgentDataTable()

            'int agentID = m_AgentID;

            Try

                If Equals(Request.Form("ctl00$MainContent$ctrAgent_Search$btnSearch"), "Search") Then ctrHiddebField.Value = Convert.ToString(m_AgentID)


                If MyBase.Page.IsPostBack And lblSearchResult.Text.Length > 0 Then
                    tblGetAgent = CType(theNotificationSystem.GetAgent(m_AgentID), NotificationSystem.NotificationSystem.Data.NotificationSystem.AgentDataTable)
                Else
                    tblGetAgent = CType(theNotificationSystem.GetAgents(), NotificationSystem.NotificationSystem.Data.NotificationSystem.AgentDataTable)
                End If


                lblSearchResult.Text = tblGetAgent.Rows.Count.ToString() & " Result(s)"
                grdAgents.DataSource = tblGetAgent.DefaultView
                grdAgents.DataBind()

            Catch __unusedException1__ As Exception
                Throw
            End Try
        End Sub



        Private Function ConvertSortDirection(e As GridViewSortEventArgs) As String
            MyBase.ViewState.Add("columnname", e.SortExpression)

            If MyBase.ViewState("direction") Is Nothing Then
                MyBase.ViewState.Add("direction", "asc")
            Else
                MyBase.ViewState("direction") = IIf(Equals(MyBase.ViewState("direction").ToString().ToLower(), "desc"), "asc", "desc")
            End If

            Return MyBase.ViewState("direction").ToString()
        End Function

        Protected Sub grdAgents_PageIndexChanging(sender As Object, e As GridViewPageEventArgs)
            Try
                If Not (IsNothing(MyBase.ViewState("columnname")) Or IsNothing(MyBase.ViewState("direction"))) Then
                    Dim m_DataView = CType(grdAgents.DataSource, DataView)

                    If m_DataView Is Nothing Then
                        m_DataView.Sort = MyBase.ViewState("columnname").ToString() & " " & MyBase.ViewState("direction").ToString()
                        grdAgents.DataSource = m_DataView
                    End If
                End If

                grdAgents.PageIndex = e.NewPageIndex
                grdAgents.DataBind()
            Catch __unusedException1__ As Exception
                Throw
            End Try
        End Sub

        Protected Sub grdAgents_Sorting(sender As Object, e As GridViewSortEventArgs)
            Try
                Dim m_Dataview = CType(grdAgents.DataSource, DataView)

                If m_Dataview Is Nothing Then
                    m_Dataview.Sort = e.SortExpression & " " & ConvertSortDirection(e)
                    grdAgents.DataSource = m_Dataview
                    grdAgents.DataBind()
                End If
            Catch __unusedException1__ As Exception
                Throw
            End Try
        End Sub
    End Class
End Namespace
